core/src/main/java/hudson/slaves/SlaveComputer.java

```diff
@@ -661,12 +669,20 @@ public void tryReconnect() {
      */
     @Deprecated
     public Slave.JnlpJar getJnlpJars(String fileName) {
-        return new Slave.JnlpJar(fileName);
+        return JnlpDataProvider.getJarFile(fileName);
     }
 
+    /**
+     * @deprecated 
+     */
     @WebMethod(name="slave-agent.jnlp")
     public HttpResponse doSlaveAgentJnlp(StaplerRequest req, StaplerResponse res) throws IOException, ServletException {
-        return new EncryptedSlaveAgentJnlpFile(this, "slave-agent.jnlp.jelly", getName(), CONNECT);
+        final HttpResponse jnlpFile = JnlpDataProvider.getJnlpFile(this, req);
```

Comment:
For a web method? No need. There should be no callers. If there are some in private plugins, those should get fixed.